Core/HLE/sceUsbMic.cpp

```diff
@@ -128,18 +128,30 @@ void __UsbMicDoState(PointerWrap &p) {
 	Do(p, curSampleRate);
 	Do(p, curChannels);
 	Do(p, micState);
+	if (s > 1) {
+		Do(p, eventUsbMicAudioUpdate);
+		CoreTiming::RestoreRegisterEvent(eventUsbMicAudioUpdate, "UsbMicAudioUpdate", &__UsbMicAudioUpdate);
+	} else {
+		eventUsbMicAudioUpdate = -1;
+	}
+
+	if (!audioBuf && numNeedSamples > 0) {
+		audioBuf = new QueueBuf(numNeedSamples << 1);
+	}
+	if(eventUsbMicAudioUpdate = -1)
+		eventUsbMicAudioUpdate = CoreTiming::RegisterEvent("UsbMicAudioUpdate", &__UsbMicAudioUpdate);
```

Comment:
You can't call RegisterEvent during DoState, it'll potentially return an id that will get allocated by something else later.

-[Unknown]

Reply:
Is there a better place to register events for old savestate?

Reply:
Oh right. Read too fast. You need to use RestoreRegisterEvent, see for example sceGe DoSTate.

Reply:
And we need to improve the internal APIs to make this kind of mistake harder to make...